src/RazorSdk/SourceGenerators/RazorDiagnostics.cs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Globalization;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.CodeAnalysis.Razor;
+
+namespace Microsoft.NET.Sdk.Razor.SourceGenerators
+{
+    internal static class RazorDiagnostics
+    {
+        public static readonly DiagnosticDescriptor InvalidRazorLangVersionDescriptor = new DiagnosticDescriptor(
+#pragma warning disable RS2008 // Enable analyzer release tracking
+            "RZ3600",
+#pragma warning restore RS2008 // Enable analyzer release tracking
+            "Invalid RazorLangVersion",
+            "Invalid value {0} for RazorLangVersion. Valid values include 'Latest' or a valid version in range 1.0 to 5.0.",
+            "Usage",
+            DiagnosticSeverity.Error,
+            isEnabledByDefault: true);
+
+        public static Diagnostic AsDiagnostic(this RazorDiagnostic razorDiagnostic)
+        {
+            var descriptor = new DiagnosticDescriptor(
+                razorDiagnostic.Id,
+                razorDiagnostic.GetMessage(CultureInfo.CurrentCulture),
+                razorDiagnostic.GetMessage(CultureInfo.CurrentCulture),
```

Comment:
This isn't the right pattern here. The compiler doesn't generate diagnostics based on `CultureInfo.CurrentCulture` but rather by the culture that's passed into the compiler options. In general you should be using `LocalizableResourceString` here. That will do the heavy lifting for you.

Reply:
https://github.com/dotnet/roslyn-analyzers/blob/5102a0d89ef178b0ca741634ee006597bf5df2ed/src/PublicApiAnalyzers/Core/Analyzers/DeclarePublicApiAnalyzer.cs#L40

Think that pattern should fit what you're doing here.

Reply:
Nice tip -- I've tracked this in https://github.com/dotnet/aspnetcore/issues/30237 since I don't think this will happen for preview2 but we can sort out the diagnostics story in preview3.